src/components/map/Map.js

```diff
@@ -23,26 +23,31 @@ const Map = () => {
   const { errorMessage, loading, results, fetchData } = useDataFetching();
   const [selectedFeature, setSelectedFeature] = useState(null);
   const [showDetailPanel, setShowDetailPanel] = useState(false);
-  const [, actions] = useAppReducer(LOCATION);
+  const [{ locations }, actions] = useAppReducer(LOCATION);
 
   const mapRef = useMap();
 
-  React.useEffect(() => {
-    (async () => {
-      const blackspotsEndpoint = `${process.env.REACT_APP_API_ROOT}blackspots/spots/?format=geojson`;
-      fetchData(blackspotsEndpoint);
-    })();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    if (locations.length === 0)
+      (async () => {
+        const blackspotsEndpoint = `${process.env.REACT_APP_API_ROOT}blackspots/spots/?format=geojson`;
```

Comment:
Wouldn't it be better to put the URL in a constants file? The previous PR had the URL written out in full

Reply:
done.